packages/arb-token-bridge-ui/src/components/TransferPanel/DepositCardL2Failure.tsx

```diff
@@ -1,14 +1,25 @@
-import { useMemo } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 
 import { MergedTransaction } from '../../state/app/state'
 import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
 import { useRedeemRetryable } from '../../hooks/useRedeemRetryable'
 import { DepositCardContainer, DepositL1TxStatus } from './DepositCard'
 import { Tooltip } from '../common/Tooltip'
 import { Button } from '../common/Button'
+import { getRetryableTicketExpiration } from 'src/util/RetryableUtils'
 
 export function DepositCardL2Failure({ tx }: { tx: MergedTransaction }) {
-  const { isConnectedToArbitrum } = useNetworksAndSigners()
+  const [retryableExpiryDays, setRetryableExpiryDays] = useState<{
+    isValid: boolean // false, if the days are still loading, or ticket is expired
+    days: number
+  }>({ isValid: false, days: 0 })
+
+  const {
+    isConnectedToArbitrum,
+    l1: { signer: l1Signer },
+    l2: { signer: l2Signer }
```

Comment:
Let's use the providers directly instead of using `signer.provider`.

Reply:
Done